test/parallel/test-tls-ticket-cluster.js

```diff
@@ -46,7 +46,9 @@ if (cluster.isPrimary) {
       session: lastSession,
       rejectUnauthorized: false
     }, () => {
-      c.end();
+      c.on('end', () => {
+        c.end();
+      });
```

Comment:
```suggestion
```

Removing the whole `'secureConnect'` listener should have the same effect.

Reply:
Thank you for your review. I applied your suggestion and tested it locally. I ran it over 4000 times and got 2 timeouts. The following is the test log:
```
E:\github\node>python tools\test.py test\parallel\test-tls-ticket-cluster
=== release test-tls-ticket-cluster ===
Path: parallel/test-tls-ticket-cluster
spawn file: E:\github\node\out\Release\node.exe
spawn file: E:\github\node\out\Release\node.exe
spawn file: E:\github\node\out\Release\node.exe
spawn file: E:\github\node\out\Release\node.exe
[primary] got "listening"
[primary] connecting 49193 session? false
[primary] got "listening"
[primary] got "listening"
[primary] got "listening"
[worker] connection reused? false
[primary] got "not-reused"
[primary] connecting 49193 session? true
[worker] connection reused? true
[primary] got "reused"
[primary] connecting 49193 session? true
[worker] connection reused? true
[primary] got "reused"
[primary] connecting 49193 session? true
[worker] connection reused? true
[primary] got "reused"
[primary] connecting 49193 session? true
[worker] connection reused? true
[primary] got "reused"
[primary] connecting 49193 session? true
[worker] connection reused? true
[primary] got "reused"
[primary] connecting 49193 session? true
[worker] connection reused? true
[primary] got "reused"
[primary] connecting 49193 session? true
[worker] connection reused? true
[primary] got "reused"
[primary] connecting 49193 session? true
[worker] connection reused? true
[primary] got "reused"
[primary] connecting 49193 session? true
[worker] connection reused? true
[primary] got "reused"
[primary] connecting 49193 session? true
[worker] connection reused? true
[primary] got "reused"
[primary] connecting 49193 session? true
[worker] connection reused? true
[primary] got "reused"
[primary] connecting 49193 session? true
[worker] connection reused? true
[primary] got "reused"
[primary] connecting 49193 session? true
[worker] connection reused? true
[primary] got "reused"
[primary] connecting 49193 session? true
[worker] connection reused? true
[primary] got "reused"
[primary] connecting 49193 session? true
[worker] connection reused? true
[primary] got "reused"
[worker] got "die"
[worker] server close
[worker] exit
[worker] got "die"
[worker] server close
[worker] exit
[worker] got "die"
[worker] server close
[worker] exit
[primary] worker died
[worker] got "die"
[worker] server close
[worker] exit
[primary] worker died
[primary] worker died
[primary] worker died
Command: E:\github\node\out\Release\node.exe E:\github\node\test\parallel\test-tls-ticket-cluster.js
--- TIMEOUT ---


[02:32|% 100|+   0|-   1]: Done

Failed tests:
E:\github\node\out\Release\node.exe E:\github\node\test\parallel\test-tls-ticket-cluster.js
```

Since the initial solution doesn't have that issue, I'd go with it to avoid timeouts.

Reply:
I think you can reproduce the timeouts also with the initial version then as the only difference is that `c.end()` is called in the next tick. See https://github.com/nodejs/node/blob/9ef03f1c237f509cb965d90c5cb301e99be80208/lib/internal/streams/readable.js#L1699

Reply:
I'm fine either way, as is or with the `'secureListener'` event listener removed. The result is the same.

Reply:
I left the test running overnight and saw the error, which is related to the handshaking. It appears that removing `secureConnect` doesn't have the same effect as adding an `end` callback. Do you have any other suggestion I could try here?
P.S. I applied your other suggestion, the code looks better now.

Reply:
> I left the test running overnight and saw the error, which is related to the handshaking. It appears that removing `secureConnect` doesn't have the same effect as adding an `end` callback. Do you have any other suggestion I could try here?

I have no suggestions. What is currently done in the `'end'` callback, is already done automatically even if the callback is not added. The only difference is that `c.end()` is called in the next tick. This makes it explicit and I'm fine with it.